### PR TITLE
Use refresh token route on unauthorized

### LINKED ISSUE #20: Use refresh_token route on Unauthorized
Try sending back the refresh_token on Unauthorized, to keep the user logged in. And implement a simple /profile page

### PR DESCRIPTION
**Pull request by desi-belokonska**
depends on resurrectism/api#24

### CODE REVIEW

File: src/api/errors.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+export class HttpServiceError extends Error {
```
Comment (sanevillain) [inline src/api/errors.ts:1]: I really like having all errors in one file (for now at least)

File: src/api/httpService.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  private static commonOptions: RequestInit = Object.freeze({
+    method: 'GET',
+    mode: 'cors',
+    headers: {
+      'Content-Type': 'application/vnd.api+json',
+    },
+    credentials: 'include',
+  });
```
Comment (sanevillain) [inline src/api/httpService.ts:43]: It would be nice to put some extra space between these two methods

**Review by sanevillain — CHANGES_REQUESTED** [review]

File: src/utils/fetchWithRefresh.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const fetchWithRefresh = (
```
Comment (sanevillain) [inline src/utils/fetchWithRefresh.ts]: I feel like this function is a bit too complicated and can be simplified by moving it to the `HttpService`

**Review by sanevillain — CHANGES_REQUESTED** [review]

File: src/utils/fetchWithRefresh.ts (hunk 3 above)
Comment (desi-belokonska) [reply src/utils/fetchWithRefresh.ts]: not sure, but I'll do it

**Review by sanevillain — APPROVED** [review]
Great. The new `Api` class looks really slim

**Pull request merged**